tools/wptrunner/wptrunner/executors/executormarionette.py

```diff
@@ -596,33 +583,13 @@ def run(self):
             self.logger.error("Lost marionette connection before starting test")
             return Stop
 
-        if timeout is not None:
-            wait_timeout = timeout + 2 * extra_timeout
-        else:
-            wait_timeout = None
-
-        timer = threading.Timer(wait_timeout, self._timeout)
-        timer.start()
-
-        self._run()
```

Comment:
I guess this is the root cause? We are stuck here forever even though the timer thread has set the flag...

Reply:
Yes.